vsintegration/src/FSharp.Editor/CodeLens/CodeLens.fs

```diff
@@ -0,0 +1,799 @@
+    let mutable bufferChangedCts = new CancellationTokenSource()
+    let uiContext = SynchronizationContext.Current
+
+    let FSharpRangeToSpan (bufferSnapshot:ITextSnapshot) (range:range) =
+        try
+            let startLine, endLine = 
+                bufferSnapshot.GetLineFromLineNumber(range.StartLine - 1),
+                bufferSnapshot.GetLineFromLineNumber(range.EndLine - 1)
+            let startPosition = startLine.Start.Add range.StartColumn
+            let endPosition = endLine.Start.Add range.EndColumn
+            Span(startPosition.Position, endPosition.Position - startPosition.Position) |> Some
+        with e -> 
+            logErrorf "Error: %A" e
+            None
+
+    let layoutTagToFormatting (layoutTag: LayoutTag) =
+        layoutTag
+        |> RoslynHelpers.roslynTag
+        |> ClassificationTags.GetClassificationTypeName
+        |> typeMap.Value.GetClassificationType
+        |> formatMap.Value.GetTextProperties   
+
+    let createTextBox (lens:CodeLens) =
+        asyncMaybe {
+            let! taggedText, navigation = lens.TaggedText
+            do! Async.SwitchToContext uiContext |> liftAsync
+            let textBox = new TextBlock(Width = 100., Background = Brushes.Transparent, Opacity = 0.5, TextTrimming = TextTrimming.WordEllipsis)
+            DependencyObjectExtensions.SetDefaultTextProperties(textBox, formatMap.Value)
+            for text in taggedText do
+                let run = Documents.Run text.Text
+                DependencyObjectExtensions.SetTextProperties (run, layoutTagToFormatting text.Tag)
+                let inl =
+                    match text with
+                    | :? Layout.NavigableTaggedText as nav when navigation.IsTargetValid nav.Range ->
+                        let h = Documents.Hyperlink(run, ToolTip = nav.Range.FileName)
+                        h.Click.Add (fun _ -> navigation.NavigateTo nav.Range)
+                        h :> Documents.Inline
+                    | _ -> run :> _
+                textBox.Inlines.Add inl
+            textBox.Opacity <- 0.5
+            lens.Computed <- true
+            lens.UiElement <- textBox
+        } |> Async.Ignore
+
+    let executeCodeLenseAsync () =  
+        asyncMaybe {
+            do! Async.Sleep 800 |> liftAsync
+            logInfof "Rechecking code due to buffer edit!"
+            let! document = workspace.CurrentSolution.GetDocument(documentId.Value) |> Option.ofObj
+            let! options = projectInfoManager.TryGetOptionsForEditingDocumentOrProject(document)
+            let! _, parsedInput, checkFileResults = checker.ParseAndCheckDocument(document, options, true, "CodeLens")
+            logInfof "Getting uses of all symbols!"
+            let! symbolUses = checkFileResults.GetAllUsesOfAllSymbolsInFile() |> liftAsync
+            let textSnapshot = buffer.CurrentSnapshot
+            logInfof "Updating due to buffer edit!"
+            
+            logInfo "HellowWorld"
```

Comment:
?

Reply:
Yaah... 😅 Debug code which shouldn't be there anymore I guess.

Reply:
and it's still there no ahaha :smile: